src/runtime_src/tools/xclbinutil/SectionBMC.cxx

```diff
@@ -36,17 +36,55 @@ SectionBMC::init::init()
 {
   auto sectionInfo = std::make_unique<SectionInfo>(BMC, "BMC", boost::factory<SectionBMC*>()); 
   sectionInfo->supportsSubSections = true;
+  sectionInfo->subSections.push_back(getSubSectionName(SubSection::FW));
+  sectionInfo->subSections.push_back(getSubSectionName(SubSection::METADATA));
+
+  // Add format support empty (no support)
+  // The top-level section doesn't support any add syntax.
+  // Must use sub-sections
+
+  sectionInfo->supportedAddFormats.push_back(FormatType::RAW);
 
   addSectionType(std::move(sectionInfo));
 }
-// -------------------------------------------------------------------------
 
-bool 
-SectionBMC::doesSupportAddFormatType(FormatType _eFormatType) const
+using SubSectionTableCollection = std::vector<std::pair<std::string, enum SectionBMC::SubSection>>;
+static const SubSectionTableCollection & 
+  getSubSectionTable() 
 {
-  // The BMC top-level section doesn't support any add syntax.  
-  // Must use sub-sections
-  return false;
+  static SubSectionTableCollection subSectionTable = {
+    std::make_pair("UNKNOWN", SectionBMC::SubSection::UNKNOWN),
+    std::make_pair("FW", SectionBMC::SubSection::FW),
+    std::make_pair("METADATA", SectionBMC::SubSection::METADATA)
+  };
+
+  return subSectionTable;
+}
+
+
+enum SectionBMC::SubSection 
+SectionBMC::getSubSectionEnum(const std::string sSubSectionName)
+{
+  auto subSectionTable = getSubSectionTable();
```

Comment:
```suggestion
  const auto& subSectionTable = getSubSectionTable();
```
There are a couple instances of this issue. I think a search for getSubSectionTable should catch all of them

Reply:
Fixed